services/galley/test/integration/API/Util.hs

```diff
@@ -2338,14 +2327,21 @@ postSSOUser name hasEmail ssoid teamid = do
 defCookieLabel :: CookieLabel
 defCookieLabel = CookieLabel "auth"
 
--- | This allows you to run requests against a galley instantiated using the given options.
---   Note that ONLY 'galley' calls should occur within the provided action, calls to other
---   services will fail.
-withSettingsOverrides :: (HasGalley m, MonadIO m, MonadMask m) => Opts.Opts -> SessionT m a -> m a
-withSettingsOverrides opts action = do
+withSettingsOverrides :: (Opts.Opts -> Opts.Opts) -> TestM a -> TestM a
+withSettingsOverrides f action = do
+  ts :: TestSetup <- ask
+  let opts = f (ts ^. tsGConf)
   (galleyApp, _, finalizer) <- liftIO $ Run.mkApp opts
```

Comment:
Since the mock server runs in `Codensity IO` anyway, it might be nicer to turn `mkApp` into a `Codensity IO` action as well, instead of returning a finaliser explicitly (which is error-prone. But I guess it's fine for now.

Reply:
I gave this a shot int https://github.com/wireapp/wire-server/pull/2381/commits/f6edc73ed9115822f495f73bf73ee0a53fc8617e I'm note sure if https://github.com/wireapp/wire-server/pull/2381/commits/92a5935a3856967ebf5fbab3bbe6635324267083 makes sense. Can you please have look again?